### PR TITLE
Take some load/save methods private

### LINKED ISSUE #225: Expose only Font and Glyph saving and loading methods
I noticed in https://github.com/linebender/norad/pull/224 that we have various public loading and saving methods that maybe don't have an audience. Exposing just the ones for Font and Glyph should be enough, and helps keeping some errors private.

### PR DESCRIPTION
**Pull request by madig**
They should probably not be used in normal operation.

Closes #225

### CODE REVIEW

**Review by cmyr — APPROVED** [review]
Thanks!

File: src/layer.rs [hunk 1]
```diff
@@ -330,7 +332,7 @@ impl Layer {
     /// [`WriteOptions`] serialization format configuration.
     ///
     /// The path should not exist.
-    pub fn save_with_options(&self, path: &Path, opts: &WriteOptions) -> Result<(), Error> {
+    pub(crate) fn save_with_options(&self, path: &Path, opts: &WriteOptions) -> Result<(), Error> {
```
Comment (chrissimpkins) [inline src/layer.rs:335]: We are using save_with_options in ufofmt.  I think you do want this pub so users can modify their write formats?

Comment (cmyr) [reply src/layer.rs:335]: are you using it on `Layer`, explicitly? My thinking was that it should be public at the `Font` level, but not for individual layers.

Comment (chrissimpkins) [reply src/layer.rs:335]: Apologies.  I shouldn't review on the mobile app... :)

Only using with Font. Thanks!

**Pull request merged**